internal/gw/publish.go

```diff
@@ -64,7 +64,13 @@ func (c *client) GetPublish(id string) Publish {
 	out.raw.Links["self"] = url
 	out.raw.Links["commit"] = url + "/commit"
 
-	return out
+	// Verify that the publish ID is valid before uploading blobs.
+	empty := struct{}{}
+	if err := c.doJSONRequest(ctx, "GET", url, nil, &empty); err != nil {
+		return out, err
+	}
+
+	return out, nil
```

Comment:
Hey team, I want to ask for a review.
1) Could you please give some advice about how to cover this line?
2) Is there any change needed in other part of this patch?

Reply:
TestClientGetPublish ought to be able to cover this line.

The fact that it currently doesn't cover this line, yet the test still somehow passes, seems to indicate the test is buggy.  I put a comment there and I'm pretty sure it would be found if you made that test check the returned error.

Have a look at `func (f *fakeGw) RoundTrip` in gw_helpers_test.go.  This function is more or less implementing a fake exodus-gw server for use in the tests.  If you have a look near the end of the function, it has some conditionals coded for the URLs it understands how to handle, like this:

`if len(route) == 1 && route[0] == "publish" && r.Method == "POST"`

Because exodus-rsync never used the "GET publish" endpoint before, there is no implementation covering it, so it falls into the default "404 not found" response.  This is the reason why the tests never reach this line of code.  If you extend the fakeGw to support getting a publish, TestClientGetPublish should be able to reach this line of code.